aws/resource_aws_appconfig_configuration_profile.go

```diff
@@ -0,0 +1,301 @@
+	retreivalRoleArn := aws.String(d.Get("retrieval_role_arn").(string))
+	if *retreivalRoleArn == "" {
+		retreivalRoleArn = nil
+	}
+
+	var validatorList []*appconfig.Validator
+	if validators := d.Get("validator").(*schema.Set).List(); len(validators) > 0 {
+		validatorList = convertMapToValidators(validators)
+	}
+
+	input := &appconfig.CreateConfigurationProfileInput{
+		ApplicationId:    applicationID,
+		Name:             configProfileName,
+		Description:      configProfileDescription,
+		LocationUri:      locationURI,
+		RetrievalRoleArn: retreivalRoleArn,
+		Validators:       validatorList,
+		Tags:             keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().AppconfigTags(),
+	}
+
+	expectedErrMsg := fmt.Sprintf("Error trying to assume role %s", d.Get("retrieval_role_arn").(string))
+	expectedErrMsg2 := "is not authorized to perform the operation"
+	var config *appconfig.CreateConfigurationProfileOutput
+	var err error
+	err = resource.Retry(iamwaiter.PropagationTimeout, func() *resource.RetryError {
+		config, err = conn.CreateConfigurationProfile(input)
+		if err != nil {
+			if isAWSErr(err, "BadRequestException", expectedErrMsg) || isAWSErr(err, "BadRequestException", expectedErrMsg2) {
+				log.Printf("[DEBUG] Retrying AppConfig Configuration Profile creation: %s", err)
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if isResourceTimeoutError(err) {
+		config, err = conn.CreateConfigurationProfile(input)
+	}
+	if err != nil {
+		return fmt.Errorf("Creating AppConfig Configuration Profile failed: %s", err)
+	}
+	log.Printf("[DEBUG] AppConfig Configuration Profile created: %s", config)
+
+	d.SetId(aws.StringValue(config.Id))
+
+	return resourceAwsAppconfigConfigurationProfileRead(d, meta)
+}
+
+func resourceAwsAppconfigConfigurationProfileRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appconfigconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	input := &appconfig.GetConfigurationProfileInput{
+		ApplicationId:          aws.String(d.Get("application_id").(string)),
+		ConfigurationProfileId: aws.String(d.Id()),
+	}
+
+	output, err := conn.GetConfigurationProfile(input)
+
+	if isAWSErr(err, appconfig.ErrCodeResourceNotFoundException, "") {
+		log.Printf("[WARN] Appconfig Configuration Profile (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error getting AppConfig Configuration Profile (%s): %s", d.Id(), err)
+	}
+
+	if output == nil {
+		return fmt.Errorf("error getting AppConfig Configuration Profile (%s): empty response", d.Id())
+	}
+
+	appARN := arn.ARN{
+		AccountID: meta.(*AWSClient).accountid,
+		Partition: meta.(*AWSClient).partition,
+		Region:    meta.(*AWSClient).region,
+		Resource:  fmt.Sprintf("application/%s/configurationprofile/%s", aws.StringValue(output.ApplicationId), aws.StringValue(output.Id)),
```

Comment:
how do you determine this?

Reply:
I looked at [this page](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsappconfig.html) for all the ARNs (look under "Resource types defined by AWS AppConfig")